packages/components/src/components/CodeEditor/useShowGutterExtension.tsx

```diff
@@ -0,0 +1,96 @@
+import { foldGutter } from "@codemirror/language";
+import { RangeSetBuilder } from "@codemirror/state";
+import {
+  EditorView,
+  gutter,
+  GutterMarker,
+  highlightActiveLine,
+  highlightActiveLineGutter,
+  lineNumbers,
+} from "@codemirror/view";
+import { useMemo } from "react";
+
+import { useReactiveExtension } from "./codemirrorHooks.js";
+
+export type EditorGutterState =
+  | {
+      type: "shown";
+      activeLineNumbers: number[];
+      onFocusFromEditorLine: (line: number) => void;
+    }
+  | { type: "hidden" };
+
+class FocusableMarker extends GutterMarker {
+  lineNumber: number;
+  onClickLine: () => void;
+
+  constructor(lineNumber: number, onClickLine: () => void) {
+    super();
+    this.lineNumber = lineNumber;
+    this.onClickLine = onClickLine;
+  }
+
+  override toDOM() {
+    const outerDiv = document.createElement("div");
+    outerDiv.className = "pr-1 cursor-pointer focusable-marker";
+
+    const marker = document.createElement("div");
+    marker.className = "focusable-marker-inner w-[2px] h-4 br-1 mt-[1px]";
+
+    outerDiv.addEventListener("click", (e) => {
+      e.preventDefault();
+      e.stopPropagation();
+      this.onClickLine();
+    });
+
+    outerDiv.appendChild(marker);
```

Comment:
This part should be doable in React. Take a look at `reactAsDom` function in `CodeEditor/utils.tsx`.